### PR TITLE
move Gnu-specific feature '--print-search-dirs' to GnuLikeCompiler

### PR DESCRIPTION
**Pull request by NNemec**
avoid a harmless but annoying error messages on non-gnu-compilers

### CODE REVIEW

File: mesonbuild/compilers/compilers.py [hunk 1]
```diff
@@ -2005,6 +2006,64 @@ def compute_parameters_with_absolute_paths(self, parameter_list, build_dir):
...
+        for p in paths:
+            # GCC returns paths like this:
+            # /usr/lib/gcc/x86_64-linux-gnu/8/../../../../x86_64-linux-gnu/lib
+            # It would make sense to normalize them to get rid of the .. parts
+            # Sadly when you are on a merged /usr fs it also kills these:
+            # /lib/x86_64-linux-gnu
+            # since /lib is a symlink to /usr/lib. This would mean
+            # paths under /lib would be considered not a "system path",
+            # which is wrong and breaks things. Store everything, just to be sure.
+            pobj = Path(p)
```
Comment (ignatenkobrain) [inline mesonbuild/compilers/compilers.py]: __[Flake8]__ [F821] undefined name 'Path' ([link](https://sider.review/gh/repos/19784232/pulls/5615#issue-4310588))

<sub>posted by Sider</sub>

File: mesonbuild/compilers/compilers.py [hunk 2]
```diff
@@ -2005,6 +2006,64 @@ def compute_parameters_with_absolute_paths(self, parameter_list, build_dir):
...
+            unresolved = pobj.as_posix()
+            if pobj.exists():
+                if unresolved not in result:
+                    result.append(unresolved)
+                try:
+                    resolved = Path(p).resolve().as_posix()
```
Comment (ignatenkobrain) [inline mesonbuild/compilers/compilers.py]: __[Flake8]__ [F821] undefined name 'Path' ([link](https://sider.review/gh/repos/19784232/pulls/5615#issue-4310589))

<sub>posted by Sider</sub>

**Comment by scivision** [comment]
Do you have an example compiler / OS where the undesired message was showing up?

**Comment by NNemec** [comment]
> Do you have an example compiler / OS where the undesired message was showing up?

I'm working with MSVC. The error about the unknown argument --print-search-dirs would show up in cases where the library would not be found anyway, so the error would never have any negative consequences. Still ugly and confusing.

To my knowledge, MSVC does not offer any alternative to this option, so we simply cannot rely on that fallback for finding libraries.

**Comment by jpakkane** [comment]
Thanks.

**Pull request merged**